### PR TITLE
CC-550 Fix: When selecting media display auto cropping for additionally selected media assets

### PR DESCRIPTION
**Pull request by aylwing-rewardstyle**
### Description: 
This PR enables the auto-cropping feature in the `YPImagePicker` library when selecting multiple media assets, automatically cropping the selected resources based on the first selected media asset size. Additionally, it resets the resources to their original state upon deactivating the selection of multiple media assets.

- Added a new `fetchImage` method for fetching original size images in `PHCachingImageManager`.
- Removed the validation that only allowed auto-cropping to videos.
- Changed the logic for resizing and centering media assets in `YPAssetZoomableView`.
- Added new properties in `YPMediaItem` for getting scale, size and crop measures.
- Change the logic for resizing videos in `YPAdjustableView`.
- Added a new container view for video assets in `YPVideoView` and `YPCoverImageView`

**Jira ticket:** [CC-550](https://rewardstyle.atlassian.net/browse/CC-550)

[CC-550]: https://rewardstyle.atlassian.net/browse/CC-550?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Source/Filters/Photo/YPCoverImageView.swift [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 import UIKit
 import Photos
+import Stevia
```
Comment (zeph-cohen10) [inline Source/Filters/Photo/YPCoverImageView.swift:10]: nit: I don't think this is strictly needed as it's imported elsewhere. Line 51 already invokes Stevia in the `setup()`

File: Source/Helpers/Extensions/PHCachingImageManager+Extensions.swift [hunk 2]
```diff
@@ -21,15 +21,29 @@ extension PHCachingImageManager {
...
-					targetSize: CGSize,
 					callback: @escaping (UIImage, [String: Any]) -> Void) {
         let options = photoImageRequestOptions()
     
         // Fetch Highiest quality image possible.
         requestImageData(for: asset, options: options) { data, _, _, _ in
             if let data = data, let image = UIImage(data: data)?.resetOrientation() {
-            
+
+                let exifs = self.metadataForImageData(data: data)
```
Comment (zeph-cohen10) [inline Source/Helpers/Extensions/PHCachingImageManager+Extensions.swift:31]: Is there a possibility for a retain cycle here? I see the `callback` is `@escaping`.  Can we capture `self` weakly?

File: Source/Pages/Gallery/YPAssetZoomableView.swift [hunk 3]
```diff
@@ -277,17 +327,22 @@ fileprivate extension YPAssetZoomableView {
...
-        var assetFrame = assetView.frame
-        let assetSize = assetView.frame.size
-        
-        assetFrame.origin.x = (assetSize.width < scrollViewBoundsSize.width) ?
-            (scrollViewBoundsSize.width - assetSize.width) / 2.0 : 0
-        assetFrame.origin.y = (assetSize.height < scrollViewBoundsSize.height) ?
-            (scrollViewBoundsSize.height - assetSize.height) / 2.0 : 0.0
-        
-        assetView.frame = assetFrame
+        DispatchQueue.main.async { [weak self] in
```
Comment (zeph-cohen10) [inline Source/Pages/Gallery/YPAssetZoomableView.swift:330]: I assume one of the call sites is invoking this in a background thread. Could we just wrap the call site in the main thread?

**Review by zeph-cohen10 — APPROVED** [review]
Changes look good 🚀

**Review by iirovi-rs — APPROVED** [review]

**Pull request merged**